test/test_datapipe.py

```diff
@@ -2752,6 +2752,17 @@ def test_sharding_length(self):
         self.assertEqual(1, len(sharded_dp0))
         self.assertEqual(0, len(sharded_dp1))
 
+    def test_sharding_groups(self):
+        numbers_dp = dp.iter.IterableWrapper(range(100))
+        sharded_dp = numbers_dp.sharding_filter()
+        sharded_dp.apply_sharding(5, 2, 0)
+        # [2, 7, 12,  17, 22, 27,  32, 37, 42,  47, 52, ...]
+        sharded_dp.apply_sharding(3, 1, 1)
+        # [7, 22, 37, 52, ...]
+        sharded_dp.apply_sharding(2, 1, 2)
+        # [22, 52, 82]
```

Comment:
nit: Is it possible to test these intermediate results as well? Maybe compared it to `list(range(2, 100, 5))`, `list(range(7, 100, 15))`, `list(range(22, 100, 30))`